src/game/hud.cpp

```diff
@@ -1758,49 +1771,58 @@ namespace hud
             {
-                int len = !full && conlines[j].type > CON_CHAT ? chatcontime/2 : chatcontime;
-                if(full || totalmillis-conlines[j].reftime <= len+chatconfade)
+                int len = !full && chatlines[j].type > CON_CHAT ? chatcontime / 2 : chatcontime;
+                if (full || totalmillis - chatlines[j].reftime <= len + chatconfade)
                 {
-                    if(refs.length() >= numl)
+                    if (refs.length() >= numl)
                     {
-                        if(refs.length() >= numo)
+                        if (refs.length() >= numo)
                         {
-                            if(full) break;
+                            if (full) break;
                             bool found = false;
-                            loopvrev(refs) if(conlines[refs[i]].reftime+(conlines[refs[i]].type > CON_CHAT ? chatcontime/2 : chatcontime) < conlines[j].reftime+len)
+                            loopvrev (refs) if (chatlines[refs[i]].reftime + (chatlines[refs[i]].type > CON_CHAT ? chatcontime / 2 : chatcontime) < chatlines[j].reftime + len)
                             {
                                 refs.remove(i);
                                 found = true;
                                 break;
                             }
-                            if(!found) continue;
+                            if (!found) continue;
                         }
-                        conlines[j].reftime = min(conlines[j].reftime, totalmillis-len);
+                        chatlines[j].reftime = min(chatlines[j].reftime, totalmillis - len);
                     }
                     refs.add(j);
                 }
             }
+
             pushhudscale(chatconscale);
-            int tx = int(x/chatconscale), ty = int(y/chatconscale),
-                ts = int(s/chatconscale), tr = tx+FONTW;
+            int tx = int(x / chatconscale), ty = int(y / chatconscale),
+                ts = int(s / chatconscale), tr = tx + FONTW;
             tz = int(tz/chatconscale);
             loopvj(refs)
             {
-                int len = !full && conlines[refs[j]].type > CON_CHAT ? chatcontime/2 : chatcontime;
-                float f = full || !chatconfade ? 1.f : clamp(((len+chatconfade)-(totalmillis-conlines[refs[j]].reftime))/float(chatconfade), 0.f, 1.f),
-                    g = conlines[refs[j]].type > CON_CHAT ? conblend : chatconblend;
-                if(chatcondate && *chatcondateformat)
-                    tz += draw_textf("%s %s", tr, ty-tz, 0, 0, 255, 255, 255, int(fade*f*g*255), TEXT_LEFT_UP, -1, ts, 1, gettime(conlines[refs[j]].realtime, chatcondateformat), conlines[refs[j]].cref)*f;
-                else tz += draw_textf("%s", tr, ty-tz, 0, 0, 255, 255, 255, int(fade*f*g*255), TEXT_LEFT_UP, -1, ts, 1, conlines[refs[j]].cref)*f;
+                int len = !full && chatlines[refs[j]].type > CON_CHAT ? chatcontime / 2 : chatcontime;
```

Comment:
* use temp vars (chatlines[ref]... > CON_CHAT)
* 1 declaration per line, separated with ';'

Reply:
I won't do that as it results in weird behavior